volatility3/framework/interfaces/objects.py

```diff
@@ -156,7 +165,7 @@ def cast(self, new_type_name: str, **additional) -> 'ObjectInterface':
         """
         # TODO: Carefully consider the implications of casting and how it should work
         if constants.BANG not in new_type_name:
-            symbol_table = self.vol['type_name'].split(constants.BANG)[0]
+            symbol_table = self.get_symbol_table_name()
```

Comment:
This one is a good catch, and we should probably be using this method more in other places.  It might be worth a grep over the rest of the code looking for a similar `.split(contants.BANG)[0]` pattern.

Reply:
Most of it is in  symbol resolution. I think that symbols should be described by an `ObjectInterface` that contains a name (IE `PsLoadedModuleList`). This would be a normalised way to access them.

Reply:
Should be a different PR though.